plugins/kotlin/idea/tests/test/org/jetbrains/kotlin/idea/presentation/DeclarationPresentersTest.kt

```diff
@@ -0,0 +1,57 @@
+// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.kotlin.idea.presentation
+
+import com.intellij.ide.IconProvider
+import com.intellij.openapi.extensions.LoadingOrder
+import com.intellij.openapi.util.Iconable
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.idea.KotlinIconProvider
+import org.jetbrains.kotlin.idea.base.fe10.codeInsight.Fe10KotlinIconProvider
+import org.jetbrains.kotlin.idea.test.KotlinLightCodeInsightFixtureTestCase
+import org.jetbrains.kotlin.psi.KtDeclaration
+import org.jetbrains.kotlin.psi.KtNamedDeclaration
+import java.awt.Component
+import java.awt.Graphics
+import javax.swing.Icon
+
+class DeclarationPresentersTest : KotlinLightCodeInsightFixtureTestCase() {
+
+    fun testGetIconWithNoExtraIconProviders() {
+        myFixture.configureByText("Foo.kt", "class <caret>Foo")
+        val element = myFixture.elementAtCaret as KtNamedDeclaration
+
+        // By default, we expect whatever the standard Fe10KotlinIconProvider returns.
+        val defaultExpectedIcon = Fe10KotlinIconProvider().getIcon(element, Iconable.ICON_FLAG_VISIBILITY or Iconable.ICON_FLAG_READ_STATUS)
```

Comment:
I'd not recommend use directly `Fe10KotlinIconProvider`, use smth like `IconProvider.EXTENSION_POINT_NAME.findFirst { it is KotlinIconProvider }`